apps/asap-server/src/handlers/webhooks/webhook-calendar-created.ts

```diff
@@ -0,0 +1,175 @@
+import AWS from 'aws-sdk';
+import Joi from '@hapi/joi';
+import { auth, JWTInput, JWT } from 'google-auth-library';
+import { framework as lambda } from '@asap-hub/services-common';
+import { WebhookPayload, Calendar } from '@asap-hub/squidex';
+import {
+  region,
+  googleApiCredentialsSecretId,
+  googleApiUrl,
+  asapApiUrl,
+} from '../../config';
+import { http } from '../../utils/instrumented-framework';
+
+import { Handler } from '../../utils/types';
+import validateRequest from '../../utils/validate-squidex-request';
+import Calendars, { CalendarController } from '../../controllers/calendars';
+
+export const webhookCalendarCreatedHandlerFactory = (
+  subscribe: SubscribeToEventChanges,
+  unsubscribe: UnsubscribeFromEventChanges,
+  calendarController: CalendarController,
+): Handler =>
+  http(
+    async (request: lambda.Request): Promise<lambda.Response> => {
+      await validateRequest(request);
+
+      const bodySchema = Joi.object({
+        type: Joi.string().required(),
+        payload: Joi.object({
+          data: Joi.object().required(),
+        })
+          .unknown()
+          .required(),
+      })
+        .unknown()
+        .required();
+
+      const { payload, type: event } = lambda.validate(
+        'body',
+        request.payload,
+        bodySchema,
+      ) as WebhookPayload<Calendar>;
+
+      if (event === 'CalendarsUpdated') {
+        if (
+          !payload.dataOld ||
+          !payload.dataOld.id ||
+          payload.dataOld.id.iv === payload.data.id.iv
+        ) {
+          return {
+            statusCode: 200,
+            payload: payload.data.id,
+          };
+        }
+
+        if (payload.dataOld.resourceId) {
+          try {
+            await unsubscribe(payload.dataOld.resourceId?.iv, payload.id);
+          } catch (error) {
+            // eslint-disable-next-line no-console
+            console.error(error);
+          }
+        }
+      }
+
+      if (['CalendarsCreated', 'CalendarsUpdated'].includes(event)) {
+        try {
+          const resourceId = await subscribe(payload.data.id.iv, payload.id);
+
+          await calendarController.update(payload.id, {
+            resourceId,
+          });
+        } catch (error) {
+          return {
+            statusCode: 400,
```

Comment:
Why would we only throw 400s?

Reply:
thats a fair point, what would you like to throw?

Reply:
If we don't want to be very specific 502 is probably the way to go.
![image](https://user-images.githubusercontent.com/6756384/108834251-dd7e9680-75c5-11eb-9597-8ca0188d85ec.png)